### PR TITLE
Fix links in concepts section

### PR DESCRIPTION
**Pull request by tengqm**
closes: #24676

This PR also fixes other link problems:

- Some links are dead because the target page has been moved;
- Some link targets are redirection records which means they only work for English version (we don't have any redirection entries for localized pages)

The link problems fixed in this PR are identified using the following command:

```
./scripts/linkchecker.py -f /docs/concepts/**/*.md
```

### CODE REVIEW

**Comment by tengqm** [comment]
FWIW, using redirection in links is bad because:

- it slows down page loading for English site, the browser has to raise a new request to the real target;
- it makes the localization teams' life difficult, they cannot simply localize the link by prefixing it with country code like 'es' or 'ko'.

File: content/en/docs/concepts/cluster-administration/system-metrics.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ The kubelet collects accelerator metrics through cAdvisor. To collect these metr
 
 The responsibility for collecting accelerator metrics now belongs to the vendor rather than the kubelet. Vendors must provide a container that collects metrics and exposes them to the metrics service (for example, Prometheus).
 
-The [`DisableAcceleratorUsageMetrics` feature gate](/docs/references/command-line-tools-reference/feature-gate.md#feature-gates-for-alpha-or-beta-features:~:text= DisableAcceleratorUsageMetrics,-false) disables metrics collected by the kubelet, with a [timeline for enabling this feature by default](https://github.com/kubernetes/enhancements/tree/411e51027db842355bd489691af897afc1a41a5e/keps/sig-node/1867-disable-accelerator-usage-metrics#graduation-criteria).
+The [`DisableAcceleratorUsageMetrics` feature gate](/docs/reference/command-line-tools-reference/feature-gates/#feature-gates-for-alpha-or-beta-features:~:text= DisableAcceleratorUsageMetrics,-false) disables metrics collected by the kubelet, with a [timeline for enabling this feature by default](https://github.com/kubernetes/enhancements/tree/411e51027db842355bd489691af897afc1a41a5e/keps/sig-node/1867-disable-accelerator-usage-metrics#graduation-criteria).
```
Comment (kbhawkey) [inline content/en/docs/concepts/cluster-administration/system-metrics.md:107]: That is an interesting link!

**Review by sftim — COMMENTED** [review]
/lgtm

File: content/en/docs/concepts/workloads/_index.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ Those workload resources include:
 * [DaemonSet](/docs/concepts/workloads/controllers/daemonset/) for running Pods that provide
   node-local facilities, such as a storage driver or network plugin;
 * [Job](/docs/concepts/workloads/controllers/job/) and
-  [CronJob](/docs/concepts/workloads/controllers/cronjob/)
+  [CronJob](/docs/concepts/workloads/controllers/cron-jobs/)
```
Comment (kbhawkey) [inline content/en/docs/concepts/workloads/_index.md:34]: OK. Wonder why this link is named "cron-jobs" vs "cronjob".

**Pull request merged**